pkg/snapshot/restorer/restorer.go

```diff
@@ -722,3 +769,87 @@ func verifySnapshotRevision(clientKV client.KVCloser, snap *brtypes.Snapshot) er
 	}
 	return nil
 }
+
+// getDecompressedSnapshotReadCloser passes the given ReadCloser through the
+// snapshot decompressor if the snapshot is compressed using a compression policy.
+// Also returns whether the snapshot was initially compressed or not, as well as
+// the compression policy used for compressing the snapshot.
+func getDecompressedSnapshotReadCloser(rc io.ReadCloser, snap *brtypes.Snapshot) (io.ReadCloser, bool, string, error) {
+	isCompressed, compressionPolicy, err := compressor.IsSnapshotCompressed(snap.CompressionSuffix)
+	if err != nil {
+		return nil, false, compressionPolicy, err
+	}
+
+	if isCompressed {
+		// decompress the snapshot
+		rc, err = compressor.DecompressSnapshot(rc, compressionPolicy)
+		if err != nil {
+			return nil, true, compressionPolicy, fmt.Errorf("unable to decompress the snapshot: %v", err)
+		}
+	}
+
+	return rc, true, compressionPolicy, nil
+}
+
+func (r *Restorer) readSnapshotContentsFromReadCloser(rc io.ReadCloser, snap *brtypes.Snapshot) ([]byte, error) {
+	startTime := time.Now()
+
+	rc, wasCompressed, compressionPolicy, err := getDecompressedSnapshotReadCloser(rc, snap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decompress delta snapshot %s : %v", snap.SnapName, err)
+	}
+
+	buf := new(bytes.Buffer)
+	bufSize, err := buf.ReadFrom(rc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse contents from delta snapshot %s : %v", snap.SnapName, err)
+	}
+
+	totalTime := time.Now().Sub(startTime).Seconds()
+	if wasCompressed {
```

Comment:
The reason to have such logs in the first place was to help with measureing the time taken for `compression/decompression of snapshots` feature when it was introduced and weigh it against the size improvement we get from comp/decomp. Since I have now changed the behavior to decompress just before applying the snapshot, I should have changed the log message as well. Thanks for pointing it out.

Reply:
The time for decompression should be measured inside:
```go
	if isCompressed {
		// decompress the snapshot
		rc, err = compressor.DecompressSnapshot(rc, compressionPolicy)
		if err != nil {
			return rc, true, compressionPolicy, fmt.Errorf("unable to decompress the snapshot: %v", err)
		}
	}
```
and therefore you do not have to pass back `wasCompressed` and `compressionPolicy` to the caller.

Reply:
if you do not wish to change this now then we can take care of this during the refactoring

Reply:
`DecompressSnapshot()` only returns the ReadCloser from the decompressor. It does not actually decompress the data immediately. The actual decompression happens only when the ReadCloser is read from (when writing to the `bytes.Buffer`). So decompression time can only be measured within the function that reads from the ReadCloser, which is `readSnapshotContentsFromReadCloser()`.